src/DataLayouts/DataLayouts.jl

```diff
@@ -590,8 +590,9 @@ end
     col[I[4]] = val
 end
 
-@inline function Base.setindex!(data::VF{S}, val, v::Integer) where {S}
-    set_struct!(view(parent(data), v, :), convert(S, val))
+@propagate_inbounds function Base.setindex!(data::VF{S}, val, v::Integer) where {S}
+    subdata = @inbounds view(parent(data), v, :)
+    set_struct!(subdata, convert(S, val))
```

Comment:
I think we can elide the view construction and just use pre-computed strides here, similar to the rewritten getstruct impl in the PR

Reply:
Yeah but this works as well xD